### PR TITLE
Add agent-inject-containers-annotation

### PR DESCRIPTION
**Pull request by HamzaZo**
Due to lack of activity in PR #163, I continued the work in this PR.

### CODE REVIEW

**Comment by HamzaZo** [comment]
Hello @tvoran 

I update the code based on your comments in #245, could you please review it ?

File: agent-inject/agent/agent.go [hunk 1]
```diff
@@ -528,6 +535,17 @@ func (a *Agent) Patch() ([]byte, error) {
 			fmt.Sprintf("/spec/containers/%d/volumeMounts", i))...)
 	}
```
Comment (tvoran) [inline agent-inject/agent/agent.go:536]: Can this loop be removed now (lines 530-536)? It looks like it's duplicated below.

Comment (HamzaZo) [reply agent-inject/agent/agent.go]: Yes, sorry for that

Comment (tvoran) [reply agent-inject/agent/agent.go]: No worries!

**Review by tvoran — APPROVED** [review]
Thanks again for pushing this forward!

**Comment by ravilr** [comment]
Hi @tvoran, now that this is released as part of `0.15.0`,  this annotation still doesn't show up in docs here: https://www.vaultproject.io/docs/platform/k8s/injector/annotations . Thanks.

**Comment by tvoran** [comment]
@ravilr Thanks, it should be there now as of https://github.com/hashicorp/vault/pull/14776.

**Pull request merged**